### PR TITLE
Ads new targeting parameter

### PR DESCRIPTION
**Pull request by andrewgeorgiou1981**
@andygnewman

### CODE REVIEW

File: ads/js/utils.js [hunk 1]
```diff
@@ -31,6 +31,23 @@ function getLayoutName () {
...
+function getVariantNames () {
+	let abState = '-';
+	const abStateEl = document.querySelector('[data-ab-state]');
+	if(abStateEl) {
+		abState = abStateEl.getAttribute('data-ab-state');
+	}
+	return abState;
+}
+
+let consolidateMetrics = function (container, metrics){
```
Comment (andygnewman) [inline ads/js/utils.js]: why use `let` rather than just `function consolidateMetrics () {}`

File: ads/js/utils.js [hunk 2]
```diff
@@ -31,6 +31,23 @@ function getLayoutName () {
...
+	let slotName = container.dataset['oAdsName'];
```
Comment (andygnewman) [inline ads/js/utils.js]: should this be a `const` rather than a `let`?

File: ads/js/utils.js [hunk 3]
```diff
@@ -31,6 +31,23 @@ function getLayoutName () {
 	return layout;
 }
 
+function getVariantNames () {
```
Comment (andygnewman) [inline ads/js/utils.js]: Perhaps would be clearer to call this function getABTestState?

File: ads/js/utils.js (hunk 2 above)
Comment (andrewgeorgiou1981) [inline ads/js/utils.js]: you're right, this can just be a function definition, rather than assign it to a var. i'll update

File: ads/index.js [hunk 4]
```diff
@@ -28,6 +28,15 @@ function initOAds (flags, appName, adOptions) {
 		const containers = [].slice.call(document.querySelectorAll('.o-ads'));
 		slotCount = containers.length;
 		utils.log.info(slotCount + ' ad slots found on page');
+		if (flags && flags.get('AdsMetricsInOneKey')) {
+			let metrics = '';
+			metrics += 'adUnit=' + initObj.gpt.site + '/' + initObj.gpt.zone;
+			metrics += (res.targeting.get().pt) ? '|pageType=' + res.targeting.get().pt : '';
+			metrics += (res.targeting.get().res) ? '|res=' + res.targeting.get().res : '';
```
Comment (andygnewman) [inline ads/index.js]: add in the mvt value

**Pull request merged**